partiql-eval/src/eval.rs

```diff
@@ -1155,7 +1156,58 @@ impl EvalExpr for EvalLikeMatch {
             Null => Value::Null,
             Missing => Value::Missing,
             Value::String(s) => Value::Boolean(self.pattern.is_match(s.as_ref())),
-            _ => Value::Boolean(false),
+            _ => Value::Missing,
+        }
+    }
+}
+
+/// Represents an evaluation `LIKE` operator without string literals in the match and/or escape
+/// pattern, e.g. in `s LIKE match_str ESCAPE escape_char`.
+#[derive(Debug)]
+pub struct EvalLikeNonStringNonLiteralMatch {
+    pub value: Box<dyn EvalExpr>,
+    pub pattern: Box<dyn EvalExpr>,
+    pub escape: Box<dyn EvalExpr>,
+}
+
+impl EvalLikeNonStringNonLiteralMatch {
+    pub fn new(
+        value: Box<dyn EvalExpr>,
+        pattern: Box<dyn EvalExpr>,
+        escape: Box<dyn EvalExpr>,
+    ) -> Self {
+        EvalLikeNonStringNonLiteralMatch {
+            value,
+            pattern,
+            escape,
+        }
+    }
+}
+
+impl EvalExpr for EvalLikeNonStringNonLiteralMatch {
```

Comment:
Added another node so the simple case of a string literal pattern (and string literal escape) (e.g. `col LIKE 'fo%`) wouldn't have to be converted to regex for every `value`